### PR TITLE
[FIX] github_connector_odoo: Repository branch modules button error

### PR DESCRIPTION
**Pull request by victoralmau**
This PR solve the problem that exists when push in "Modules" button from Repository branch menu.

Please, @pedrobaeza review it

@Tecnativa TT25565

### CODE REVIEW

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi. If you remove this part, the button will display all the module.version and not only versions of the current branch. 
i think you should introduce a new action action_repository_branch_2_module_version with a context. 

don't you think ?

**Comment by pedrobaeza** [comment]
@victoralmau The idea is to replace the `<button>` with action by a `<button>` with a method (`type="object"`) that reads the action and pass the proper id by context instead of using `active_id`.

**Comment by legalsylvain** [comment]
Hi @pedrobaeza : do you see it's better to have an action type="object" with a new function, or just a new action like this one ? 
https://github.com/odoo/odoo/blob/12.0/addons/sale/views/res_partner_views.xml#L3

something like : 

    <record id="action_repository_branch_2_module_version" model="ir.actions.act_window">
            <field name="name">Module Versions<field>
            <field name="res_model">odoo.module.version</field>
            <field name="view_type">form</field>
            <field name="view_mode">tree,form</field>
            <field name="context">{'search_default_repository_branch_id': active_id}</field>
     </record>

Just a question / no blocking point of course.

**Comment by pedrobaeza** [comment]
Yes, we switched most of them (this one seems a leftover), as the problem is the following:

- You access for example repository.
- Navigate through smart-button to branch.
- Then navigate through smart-button to module versions.

The `active_id` is the one from repository, not branch! It's an Odoo fault, but we workaround this way. It's this issue: #28

**Comment by legalsylvain** [comment]
Ow ! I didn't know. That's very bad !
Thanks for the information !

**Comment by victoralmau** [comment]
Sorry, now the button is change to object and create the action in the model that work fine.

**Review by legalsylvain — APPROVED** [review]
thanks.

**Comment by pedrobaeza** [comment]
I'm seeing the same action in other buttons, so potentially they will have the same problem. Can you please turn all of them to this system, trying to optimize the code?

**Comment by victoralmau** [comment]
> I'm seeing the same action in other buttons, so potentially they will have the same problem. Can you please turn all of them to this system, trying to optimize the code?

Changes apply to all buttons type=action

File: github_connector_odoo/models/odoo_module.py [hunk 1]
```diff
@@ -178,3 +178,23 @@ def create_if_not_exist(self, technical_name):
...
+        action["context"]["search_default_module_id"] = self.id
+        return action
+
+    def action_odoo_module_version_dependency(self):
+        self.ensure_one()
+        action = self.env.ref(
+            "github_connector_odoo.action_odoo_module_version"
+        ).read()[0]
+        action["context"] = dict(self.env.context)
+        action["context"].pop("group_by", None)
```
Comment (joao-p-marques) [inline github_connector_odoo/models/odoo_module.py]: Both the `action_odoo_module_version()` and `action_odoo_module_version_dependency()` functions share the same logic, with exception for the final `search_*` assignment. Would it be possible to move the shared logic into a 3rd function that both of them call?

Comment (pedrobaeza) [reply github_connector_odoo/models/odoo_module.py]: I was thinking more in the following: having a mixin that add a generic method `action_open` that passes 3 arguments:

- XML-ID of the action to read.
- Name of the field to look for (then you add later the `search_default` prefix).
- ID of the record for that field.

Put the logic that way.

Then inherit all models from that mixin and call the method from each button with the proper arguments.

Comment (joao-p-marques) [reply github_connector_odoo/models/odoo_module.py]: Yes, inheritance seems like a better option, indeed. I just didn't know if we could share the logic between all the models, but would certainly make it a lot more maintainable :+1: .

Comment (victoralmau) [reply github_connector_odoo/models/odoo_module.py]: OK, thanks, good suggestion. I work in this change.

File: github_connector_odoo/models/odoo_license.py [hunk 2]
```diff
@@ -51,3 +52,9 @@ def create_if_not_exist(self, name):
         if not module:
             module = self.create({"name": name})
         return module
+
+    def action_open(self):
```
Comment (pedrobaeza) [inline github_connector_odoo/models/odoo_license.py]: Good idea about the context, but put this method also on the mixin

File: github_connector_odoo/models/odoo_lib_python.py [hunk 3]
```diff
@@ -41,3 +42,9 @@ def create_if_not_exist(self, name):
         if not lib_python:
             lib_python = self.create({"name": name})
         return lib_python
+
+    def action_open(self):
+        self.ensure_one()
+        return super().action_open(
+            self._context.get("xml_id"), self._context.get("name"), self.id
```
Comment (pedrobaeza) [inline github_connector_odoo/models/odoo_lib_python.py]: Better to be `field_name` instead of `name`

**Review by joao-p-marques — APPROVED** [review]

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please remove the submodule interface-github

**Comment by victoralmau** [comment]
> Please remove the submodule interface-github

Sorry, solved

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**